satellite/metainfo/batch.go

```diff
@@ -19,9 +22,29 @@ func (endpoint *Endpoint) Batch(ctx context.Context, req *pb.BatchRequest) (resp
 
 	resp.Responses = make([]*pb.BatchResponseItem, 0, len(req.Requests))
 
-	// TODO find a way to pass some parameters between request -> response > request
-	// TODO maybe use reflection to shrink code
+	var lastResponse *pb.BatchResponseItem
+	var lastStreamID storj.StreamID
+	var lastSegmentID storj.SegmentID
 	for _, request := range req.Requests {
+
+		if lastResponse != nil {
+			responseElem := reflect.ValueOf(lastResponse.Response).Elem()
```

Comment:
Cannot we write the lastStreamID/SegmentID after every call?

Reply:
then something like `Batch(ListSegments, DeleteSegment, DeleteSegment, DeleteSegment)` wont work because second request will change SegmentID.